lib/scope.js

```diff
@@ -176,16 +176,27 @@ function defineScope(cls, targetClass, name, params, methods, options) {
           }
         }
       };
-      
+
       f._receiver = this;
       f._scope = typeof definition.params === 'function' ?
         definition.params.call(self) : definition.params;
-      
+
       f._targetClass = targetModel.modelName;
       if (f._scope.collect) {
         f._targetClass = i8n.camelize(f._scope.collect);
       }
-      
+
+      f.getAsync = function (cond, cb) {
+        if (cb === undefined && cond === undefined) {
+          cb = utils.createPromiseCallback();
+          cond = true;
+        } else if (cb === undefined && typeof cond !== 'function') {
+          cb = utils.createPromiseCallback();
+        }
+        definition.related(self, f._scope, cond, cb);
```

Comment:
I think this code does not correctly handle the variant `getAsync(mycb)`. 

It will call `related(self, f._scope, mycb, undefined)`, and because `related` is checking `arguments.length`, it will parse the arguments incorrectly.

I think this is the correct implementation:

```js
  if (cb === undefined) {
    if (cond === undefined) {
      // getAsync()
      cb = utils.createPromiseCallback();
      cond = true;
    } else if (typeof cond !== 'function') {
      // getAsync({where:{}})
      cb = utils.createPromiseCallback();
    } else {
      // getAsync(function(){})
      cb = cond;
      cond = true;
    }
  }
```

Thoughts?